### PR TITLE
neetoui v2 migration

### LINKED ISSUE #724: Migrate to neetoui-v2
@edwinbbu _a

cc @karthiknmenon @yedhink

### PR DESCRIPTION
**Pull request by edwinbbu**
Fixes #724 

- Migrated all pages and layout to v2
- Updated toastify package
- Update note support added
- Folder restructure and code improvements

### CODE REVIEW

**Comment by edwinbbu** [comment]
@karthiknmenon @yedhink _a please review

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      } else {
+        await notesApi.create(values);
+      }
+      refetch();
+      onClose();
+    } catch (err) {
+      logger.error(err);
+    }
+  };
+  return (
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:24]:
```suggestion
  };
  
  return (
```

**Review by yedhink — COMMENTED** [review]
@edwinbbu _a

I have added some comments. Please go through them and let me know if you need any clarifications.

File: app/controllers/api/v1/notes_controller.rb [hunk 2]
```diff
@@ -21,6 +21,18 @@ def create
     end
   end
 
+  def update
+    if @note.update(note_params)
+      render json: {
+        notice: "#{@note.title.humanize} note has been updated"
+      }
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/controllers/api/v1/notes_controller.rb]: Done

File: app/controllers/api/v1/notes_controller.rb [hunk 3]
```diff
@@ -21,6 +21,18 @@ def create
...
+    else
+      render json: {
+        error: note.errors.full_messages.to_sentence
+      }, status: :unprocessable_entity
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/controllers/api/v1/notes_controller.rb]: Done

File: app/controllers/api/v1/notes_controller.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class Api::V1::NotesController < Api::V1::BaseController
-  before_action :load_note!, only: [:show, :delete]
+  before_action :load_note!, only: [:update, :delete]
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/controllers/api/v1/notes_controller.rb]: Done

File: app/javascript/src/components/Dashboard/Notes/Pane/CreateNote.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+import { Pane, Typography } from "neetoui/v2";
+
+import formInitialValues from "constants/formInitialValues";
+
+import Form from "./Form";
+
+export default function NewNotePane({ fetchNotes, showPane, setShowPane }) {
+  const onClose = () => setShowPane(false);
+  return (
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/javascript/src/components/Dashboard/Notes/Pane/CreateNote.jsx:11]: Done

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx (hunk 1 above)
Comment (edwinbbu) [reply app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:24]: Done

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    try {
+      if (isEdit) {
+        await notesApi.update(note.id, values);
+      } else {
+        await notesApi.create(values);
+      }
+      refetch();
+      onClose();
+    } catch (err) {
+      logger.error(err);
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:21]: Done. Changed to toastr.

File: app/javascript/src/components/Dashboard/Notes/Pane/EditNote.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import Form from "./Form";
+
+export default function EditNotePane({
+  fetchNotes,
+  showPane,
+  setShowPane,
+  note,
+}) {
+  const onClose = () => setShowPane(false);
+  return (
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/javascript/src/components/Dashboard/Notes/Pane/EditNote.jsx:14]: Done

File: app/javascript/src/components/Common/Sidebar/constants.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+
+const NotesIcon = () => <i className="text-2xl ri-file-text-line" />;
+const SettingsIcon = () => <i className="text-2xl ri-settings-2-line" />;
+
+export default {
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/javascript/src/components/Common/Sidebar/constants.js:6]: Changed to `export const`. I just referred from other place. `export const` is the better way.

File: app/javascript/src/components/Common/Sidebar/index.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        subdomain: "bigbinary.com",
+      }}
+      profileInfo={{
+        name: `${user.first_name} ${user.last_name}`,
+        imageUrl: user.profile_image_path,
+        email: user.email,
+        dropdownProps,
+      }}
+      footerLinks={constants.FOOTER_LINKS}
+      onCollapse={() => setIsSidebarCollapsed(!isSidebarCollapsed)}
```
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/javascript/src/components/Common/Sidebar/index.js:64]: It is fine. We are having animation for sidebar which takes time to open and close. Hence we don't have use prevState.
Always using prevState is not a best practice.

**Comment by edwinbbu** [comment]
@yedhink _a Please review.

File: app/controllers/api/v1/notes_controller.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (edwinbbu) [reply app/controllers/api/v1/notes_controller.rb]: Is there a specific reason we are using `%i` notation. In many places, it's the other way.

[the remark this answers is not in the document]
Comment (yedhink) [reply app/controllers/api/v1/notes_controller.rb]: @edwinbbu 

Please read [this section](https://www.bigbinary.com/learn-rubyonrails-book/adding-a-new-task-using-create-action#:~:text=If%20you%20are%20wondering,w%5BRuby%20Python%20PHP%5D.) from our LRRB.

**Review by yedhink — COMMENTED** [review]
@edwinbbu _a

One minor comment added. Rest LGTM.

You can assign this PR to Karthik so that he could verify UI changes. Then I think we can merge it.

**Comment by edwinbbu** [comment]
@karthiknmenon _a Please review

**Review by yedhink — APPROVED** [review]

**Review by karthiknmenon — APPROVED** [review]

File: app/controllers/api/v1/notes_controller.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (nithinkr1shna) [reply app/controllers/api/v1/notes_controller.rb]: JFYI: http://kenglish.co/blog/2020/5/rubisms/

**Pull request merged**